pdstable/pds4table.py

```diff
@@ -0,0 +1,331 @@
+            except:
+                raise IOError('Table file name was not found in PDS4 label')
+
+            if table_file is None or table_file not in table_name_li:
+                raise ValueError(f"The table file name '{table_file}' doesn't exist. " +
+                                 f'The label contains {len(table_name_li)} table ' +
+                                 f'files: {table_name_li}')
+            else:
+                self.table_file_name = table_file
+                self.table_file_li = table_name_li
+                # specify the table file that we want to read
+                idx = table_name_li.index(table_file)
+                file_area = file_area[idx]
+        # The label file has no table file info
+        else:
+            raise ValueError(f'{label_file_path} does not contain any table file info.' )
+
+        try:
+            self.header_bytes = int(file_area['Header']['object_length'])
+        except KeyError:
+            # Some tables don't have header
+            self.header_bytes = 0
+
+        # table info is the last child tag of the file area
+        table_area = file_area[list(file_area.keys())[-1]]
+        self.rows = int(table_area['records'])
+        # record info is the last child tag of the table info area
+        record_area = table_area[list(table_area.keys())[-1]]
+        self.columns = int(record_area['fields'])
+
+        try:
+            # for the table with fixed row length
+            self.row_bytes = int(record_area['record_length'])
+            self.fixed_length_row = True
+            self.field_delimiter = None
+        except:
+            # for the case like .csv table, row length is not used
+            self.row_bytes = int(record_area['maximum_record_length'])
+            self.fixed_length_row = False
+            self.field_delimiter = PDS4_FIELD_DELIMITER[table_area['field_delimiter']]
+
+        # Save the key info about each column in a list and a dictionary
+        self.column_info_list = []
+        self.column_info_dict = {}
+
+        # Construct the dtype0 dictionary
+        self.dtype0 = {'crlf': ('|S2', self.row_bytes-2)}
+
+        default_invalid = set(invalid.get('default', []))
+
+        # Get all the columns info from the record tags in the xml file
+        # columns info is the last child tag of the record info area
+        columns = record_area[list(record_area.keys())[-1]]
+        for col in columns:
+            name = col['name']
+            field_num = int(col['field_number'])
+
+            pdscol = Pds4ColumnInfo(col, field_num,
+                        invalid = invalid.get(name, default_invalid),
+                        valid_range = valid_ranges.get(name, None))
+
+            # PDS4 TODO: Do we have more duplicated column names, except the 'Target'?
+            # Handle duplicated column name 'Target'
+            if name in self.column_info_dict and name != 'Target':
```

Comment:
There's nothing special about "Target". The creator of the Uranus index table just decided to give two columns the same name. I don't think this was a great idea, but I don't think it's technically illegal. Talk to @markshowalter about how to handle this case. Personally I would do something like number them `Target_1` and `Target_2`.